raiden/network/proxies/secret_registry.py

```diff
@@ -88,61 +120,175 @@ def register_secret_batch(
+                    # can happen because of:
+                    #
+                    # - A compiler bug if a revert was introduced.
+                    # - A configuration bug, because for 0.4.0 the secret
+                    # registry does not have a revert.
+                    error = (
+                        'Secret registration failed because of a configuration '
+                        'bug or compiler bug. Please double check the secret '
+                        'smart contract is at version 0.4.0, if it is then a '
+                        'compiler bug was hit.'
+                    )
+
+                log.critical(error, **log_details)
+                exception = RaidenUnrecoverableError(error)
+                transaction_result.set_exception(exception)
+                raise exception
+
+            # If gas_limit is set and there is no receipt then an exception was
+            # raised while sending the transaction. This should only happen if
+            # the account is being used concurrently, which is not supported.
+            # This can happen because:
+            #
+            # - The nonce of the transaction was already used
+            # - The nonce was reused *and* the account didn't have enough ether
+            # to pay for the gas
+            #
+            # Safety cannot be guaranteed under any of these cases, this error
+            # is unrecoverable. *Note*: This assumes the ethereum client
+            # takes into account the current transactions in the pool.
+            if gas_limit:
+                assert msg, (
+                    'Unexpected control flow, an exception should habe been raised.'
+                )
+                error = (
+                    f"Sending the the transaction for registerSecretBatch failed with: `{msg}`. "
+                    f"This happens if the same ethereum account is being used by more than one "
+                    f"program which is not supported."
+                )
 
+                log.critical(error, **log_details)
+                exception = RaidenUnrecoverableError(error)
+                transaction_result.set_exception(exception)
+                raise exception
+
+            # gas_limit can fail because:
+            #
+            # - The Ethereum client detected the transaction could not
+            # successfully execute, this happens if an assert/revert is hit.
+            # - The account is lacking funds to pay for the gas.
+            #
+            # Either of these is a bug. The contract does not use
+            # assert/revert, and the account should always be funded
+
+            # For the
             self.proxy.jsonrpc_client.check_for_insufficient_eth(
                 transaction_name='registerSecretBatch',
-                transaction_executed=transaction_executed,
-                required_gas=len(secrets) * GAS_REQUIRED_PER_SECRET_IN_BATCH,
```

Comment:
Again, same problem as before, now it just uses the same `gas_limit` that failed the transaction